src/coreclr/jit/emitxarch.cpp

```diff
@@ -5963,6 +5970,19 @@ bool emitter::HasSideEffect(instruction ins, emitAttr size)
             break;
         }
 
+        case INS_movdqa32:
+        case INS_movdqa64:
+        case INS_movdqu8:
+        case INS_movdqu16:
+        case INS_movdqu32:
+        case INS_movdqu64:
+        {
+            // These EVEX instructions merges/masks based on k-register
+            // TODO-XArch-AVX512 : Handle merge/masks scenarios once k-mask support is added for these.
+            hasSideEffect = UseEvexEncoding() && (size < EA_64BYTE);
```

Comment:
This can probably be:
```suggestion
            assert(UseEvexEncoding());
            hasSideEffect = (size != EA_64BYTE);
```

Reply:
Changed